src/com/CommunicateBoundingBox.cpp

```diff
@@ -54,6 +50,7 @@ void CommunicateBoundingBox::receiveBoundingBoxMap(
   int sizeOfReceivingMap;
   _communication->receive(sizeOfReceivingMap, rankSender);
 
+  PRECICE_CHECK(sizeOfReceivingMap == (int) bbm.size(), "Incoming size of map is not compatible");
   PRECICE_ASSERT(sizeOfReceivingMap == (int) bbm.size());
```

Comment:
```suggestion
  PRECICE_ASSERT(sizeOfReceivingMap == (int) bbm.size(), "Incoming size of map is not compatible");
```
This should be an assertion. A user of preCICE should not be able to reach that state. It's merely to safeguard against bugs in preCICE.